src/main/java/backend/Logic.java

```diff
@@ -247,13 +209,204 @@ public Model getRepo(String repoId) {
+                List<TurboIssue> issuesInRepo = toUpdate.get(issue.getRepoId());
+                if (issuesInRepo != null) {
+                    // If eixsts, just add the issue
+                    issuesInRepo.add(issue);
+                } else {
+                    // If not exists, create first then add
+                    issuesInRepo = new ArrayList<>();
+                    issuesInRepo.add(issue);
+                    toUpdate.put(issue.getRepoId(), issuesInRepo);
+                }
+            });
+        });
+
+        return toUpdate;
+    }
+
+    /**
+     * Retrieves metadata for given issues from the repository source, and then processes them for non-self
+     * update timings.
+     *
+     * @param repoId The repository containing issues to retrieve metadata for.
+     * @param issues Issues sharing the same repository requiring a metadata update.
+     * @return True if metadata retrieval was a success, false otherwise.
+     */
+    public CompletableFuture<Boolean> getIssueMetadata(String repoId, List<TurboIssue> issues) {
+        String message = "Getting metadata for " + repoId + "...";
+        logger.info("Getting metadata for issues "
+                + issues.stream().map(TurboIssue::getId).map(Object::toString).collect(Collectors.joining(", ")));
+        UI.status.displayMessage(message);
+
+        String currentUser = prefs.getLastLoginUsername();
+
+        return repoIO.getIssueMetadata(repoId, issues).thenApply(this::processUpdates)
+                .thenApply(metadata -> {
+                    String updatedMessage = "Received metadata from " + repoId + "!";
+                    UI.status.displayMessage(updatedMessage);
+                    models.insertMetadata(repoId, metadata, currentUser);
+                    return true;
+                })
+                .exceptionally(withResult(false));
+    }
+
+    /**
+     * Filters and sorts issues within the model according to the given filter expressions.
+     *
+     * @param filterTexts Filter expressions to process.
+     * @return Filter expressions and their corresponding issues after filtering and sorting.
+     */
+    private Map<FilterExpression, ImmutablePair<List<TurboIssue>, Boolean>>
+                                        filterAndSort(List<FilterExpression> filterTexts) {
+        Map<FilterExpression, ImmutablePair<List<TurboIssue>, Boolean>> filteredAndSorted = new HashMap<>();
+
+        List<TurboIssue> allModelIssues = models.getIssues();
+
+        filterTexts.forEach(filterText -> {
+            ImmutablePair<List<TurboIssue>, Boolean> filterAndSortedExpression = filteredAndSorted.get(filterText);
+
+            if (filterAndSortedExpression == null) { // If it already exists, no need to filter anymore
+                List<Qualifier> metaQualifiers = filterText.find(Qualifier::isMetaQualifier);
+                boolean hasUpdatedQualifier = Qualifier.hasUpdatedQualifier(metaQualifiers);
```

Comment:
`metaQualifiers` is used in the `determineComparator` call. Maybe I should just pass the filter expression itself to `determineComparator`?